lib/widgets/book_list.dart

```diff
@@ -0,0 +1,99 @@
+import 'package:flutter/material.dart';
+import '../models/book.dart';
+
+class BookListWidget extends StatelessWidget {
+  final String title;
+  final List<Book> books;
+  const BookListWidget({
+    required this.title,
+    required this.books,
+    Key? key,
+  }) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Text(
+          title,
+          style: const TextStyle(
+            color: Colors.black,
+            fontSize: 30,
+            fontWeight: FontWeight.bold,
+          ),
+        ),
+        const SizedBox(height: 10),
+        SizedBox(
+          height: 250,
+          child: ListView.builder(
+            physics: const ClampingScrollPhysics(),
+            shrinkWrap: true,
+            scrollDirection: Axis.horizontal,
+            itemCount: books.length,
+            itemBuilder: (BuildContext context, int index) {
+              final book = books[index];
+              return Column(
+                children: [
+                  SizedBox(
+                    width: 105,
+                    height: 220,
+                    child: Card(
+                      borderOnForeground: false,
+                      elevation: 0,
+                      child: Column(
+                        children: [
+                          SizedBox(
+                            width: 97,
+                            height: 170,
+                            child: ClipRRect(
+                              borderRadius: BorderRadius.circular(10),
+                              child: Image.network(
+                                book.coverUrl,
+                                fit: BoxFit.cover,
+                              ),
+                            ),
+                          ),
+                          const SizedBox(
+                            height: 10,
+                          ),
+                          SizedBox(
+                            height: 30, //[MODIFY]
+                            width: 97,
+                            child: Column(
+                              crossAxisAlignment: CrossAxisAlignment.start,
+                              children: [
+                                Text(
+                                  book.title.length > 10
+                                      ? '${book.title.substring(0, 9)}...'
+                                      : book.title,
```

Comment:
You also can set the maxiLine for the Text widget. If the text gets longer for a given area, the "..." will be shown.

```dart
Text(
  'A long sentence',
  maxLines: 1,
)
```

Reply:
Thank you, That really helpful.